ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/math/Greatest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.math;
+
+import com.google.common.collect.Streams;
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.function.udf.UdfSchemaProvider;
+import io.confluent.ksql.schema.ksql.SqlArgument;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.DecimalUtil;
+import io.confluent.ksql.util.KsqlConstants;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+@UdfDescription(
+    name = "greatest",
+    category = FunctionCategory.MATHEMATICAL,
+    description = "Returns the highest value among a variable number of comparable columns",
```

Comment:
Non-null elements only, right? (Same for Least.java below.)

Reply:
It can accept null values, but will only return null if all input values are null. "Returns the highest value among a variable number of non-null compable columns" implies to me that we will outright disallow null input, which is untrue. Same issue as above, in a way. Any input on the best way to communicate this is appreciated.

Reply:
I was thinking "Returns the highest non-null value among a variable number of comparable columns." If you want to be super precise we could say "Returns the highest non-null value (if any) among a variable number of comparable columns" but I don't feel strongly one way or the other.

Reply:
I think I prefer the former to make it a little more concise. I don't envision a world where the user is so confused by this that it'd cause a problem they couldn't solve with a 2 minute test.